File: wrappers/java/spring-demos/src/main/java/com/kendoui/spring/controllers/waterfallcharts/LocalDataBindingController.java
package com.kendoui.spring.controllers.waterfallcharts;

import java.util.ArrayList;
import java.util.List;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import com.kendoui.spring.models.CashFlowData;

@Controller("dataviz-waterfall_chart-local_data-controller")
@RequestMapping(value="/waterfall-charts/")
public class LocalDataBindingController {
    @RequestMapping(value = "/local-data-binding", method = RequestMethod.GET)
    public String index(Model model) {
        List<CashFlowData> viewModel = new ArrayList<CashFlowData>();       
        viewModel.add(new CashFlowData("Beginning\nBalance", 50000));
        viewModel.add(new CashFlowData("Jan", 17000));
        viewModel.add(new CashFlowData("Feb", 14000));
        viewModel.add(new CashFlowData("Mar", -12000));
        viewModel.add(new CashFlowData("Q1", "runningTotal"));
        viewModel.add(new CashFlowData("Apr", -22000));
        viewModel.add(new CashFlowData("May", -18000));
        viewModel.add(new CashFlowData("Jun", 10000));
        viewModel.add(new CashFlowData("Q2", "runningTotal"));
        viewModel.add(new CashFlowData("Ending\nBalance", "total"));

        model.addAttribute("viewModel", viewModel);
     
        return "/waterfall-charts/local-data-binding";
    }
}

File: wrappers/mvc/demos/Kendo.Mvc.Examples/Controllers/Waterfall_Charts/Local_Data_BindingController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using Kendo.Mvc.Examples.Models;

namespace Kendo.Mvc.Examples.Controllers
{
    public partial class Waterfall_ChartsController : Controller
    {
        public ActionResult Local_Data_Binding()
        {
            var model = new CashFlowData[] {
                new CashFlowData
                {
                    Period = "Beginning\nBalance",
                    Amount = 50000
                },
                new CashFlowData
                {
                    Period = "Jan",
                    Amount = 17000
                },
                new CashFlowData 
                {
                    Period = "Feb",
                    Amount = 14000
                },
                new CashFlowData
                {
                    Period = "Mar",
                    Amount = -12000
                },
                new CashFlowData
                {
                    Period = "Q1",
                    Summary = "runningTotal"
                },
                new CashFlowData
                {
                    Period = "Apr",
                    Amount = -22000
                },
                new CashFlowData
                {
                    Period = "May",
                    Amount = -18000
                },
                new CashFlowData
                {
                    Period = "Jun",
                    Amount = 10000
                },
                new CashFlowData
                {
                    Period = "Q2",
                    Summary = "runningTotal"
                },
                new CashFlowData
                {
                    Period = "Ending\nBalance",
                    Summary = "total"
                }
            };

            return View(model);
        }
    }
}